packages/modules-sdk/src/utils/clean-graphql-schema.ts

```diff
@@ -0,0 +1,81 @@
+import { Kind, parse, print, visit } from "graphql"
+
+export function cleanGraphQLSchema(schema: string): {
+  schema: string
+  notFound: Record<string, Record<string, string>>
+} {
+  const extractTypeName = (type) => {
+    if (type.kind === Kind.NAMED_TYPE) {
+      return type.name.value
+    }
+    if (type.kind === Kind.NON_NULL_TYPE || type.kind === Kind.LIST_TYPE) {
+      return extractTypeName(type.type)
+    }
+    return null
+  }
+
+  const ast = parse(schema)
+
+  const typeNames = new Set(["String", "Int", "Float", "Boolean", "ID"])
+  const extendedTypes = new Set()
+
+  const kinds = [
+    Kind.OBJECT_TYPE_DEFINITION,
+    Kind.INTERFACE_TYPE_DEFINITION,
+    Kind.ENUM_TYPE_DEFINITION,
+    Kind.SCALAR_TYPE_DEFINITION,
+    Kind.INPUT_OBJECT_TYPE_DEFINITION,
+    Kind.UNION_TYPE_DEFINITION,
+  ]
+  ast.definitions.forEach((def: any) => {
+    if (kinds.includes(def.kind)) {
+      typeNames.add(def.name.value)
+    } else if (def.kind === Kind.OBJECT_TYPE_EXTENSION) {
+      extendedTypes.add(def.name.value)
+    }
+  })
+
+  const nonExistingMap: Record<string, Record<string, string>> = {}
+  const parentStack: string[] = []
+
+  const cleanedAst = visit(ast, {
```

Comment:
comment: looks very clean; could you add a couple of comments in code here, describing the work being done? Think that would be helpful for the future